Sources/TSCBasic/Path.swift

```diff
@@ -534,36 +513,17 @@ private struct WindowsPath: Path, Sendable {
         return String(cString: representation)
     }
 
-    init(normalizingAbsolutePath path: String) {
-        self.init(string: Self.repr(path).withCString(encodedAs: UTF16.self) { pwszPath in
-          var canonical: PWSTR!
-          _ = PathAllocCanonicalize(pwszPath,
-                                    ULONG(PATHCCH_ALLOW_LONG_PATHS.rawValue),
-                                    &canonical)
-          return String(decodingCString: canonical, as: UTF16.self)
-        })
-    }
-
     init(validatingAbsolutePath path: String) throws {
-        let realpath = Self.repr(path)
+        let fsr: UnsafePointer<Int8> = path.fileSystemRepresentation
+        defer { fsr.deallocate() }
+
+        let realpath = String(cString: fsr)
         if !Self.isAbsolutePath(realpath) {
             throw PathValidationError.invalidAbsolutePath(path)
         }
         self.init(normalizingAbsolutePath: path)
     }
 
-    init(normalizingRelativePath path: String) {
-        if path.isEmpty || path == "." {
-            self.init(string: ".")
-        } else {
-            var buffer: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(MAX_PATH + 1))
-            _ = path.replacingOccurrences(of: "/", with: "\\").withCString(encodedAs: UTF16.self) {
-                PathCanonicalizeW(&buffer, $0)
-            }
-            self.init(string: String(decodingCString: buffer, as: UTF16.self))
-        }
-    }
-
```

Comment:
Restoring this makes things build again, was this intentionally removed?

Reply:
sure, is this windows only?

Reply:
for reference, it was removed because hit was removed from the Path API - the whole idea you pushed for was  that we dont normalize / canonicalize  on construction because that cant work correctly on windows

Reply:
perhaps WindowsPath `init(validatingAbsolutePath)` and `init(validatingRelativePath)` should call `self.init(string:)` instead of `self.init(normalizingAbsolutePath:)`?